bin/run-circleci-tests

```diff
@@ -4,11 +4,34 @@ set -e
 
 make install_system_dependencies
 
-case $CIRCLE_NODE_INDEX in
-    0)
-        make test
-        ;;
-    *)
-        make test_integration
-        ;;
-esac
+# Try to get Redis up and running before launching tests.
+# In the past, individual tests from our unit test suite would sometimes fail with a "ConnectionRefusedError"
+# that was caused by Redis being unavailable in the container running the tests.
+# The solution below is based on suggestions from the following support thread:
+# https://discuss.circleci.com/t/redis-server-problems-unable-to-connect-localhost-6379/4040
+for i in {1..300}; do
+    if [ bash -c "sudo service redis-server status" ]
```

Comment:
@itsjeyd Build fails on this line. I'm not really a bash expert, but I guess that `[`is redundant here. (`if` operator takes a command and check exit code, and `[` is really a alias to `test` command).